modules/gitrepo/commit.go

```diff
@@ -0,0 +1,26 @@
+// Copyright 2025 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package gitrepo
+
+import (
+	"context"
+
+	"code.gitea.io/gitea/modules/git"
+)
+
+func GetCommit(ctx context.Context, repo Repository, commitID string) (*git.Commit, error) {
+	gitRepo, err := git.OpenRepository(ctx, repoPath(repo))
+	if err != nil {
+		return nil, err
+	}
+	defer gitRepo.Close()
+	return gitRepo.GetCommit(commitID)
```

Comment:
You can not do this! You can not close the gitRepo then then access the returned "commit"!!!!

That's I said many times: I do not see how you could make the "gitrepo" package work correctly, but introducing more bugs.

Reply:
998f9ff